File: algorithms/sorting-algorithms/bubble-sort/bubble_sort.py
arr = [8,2,4,3]

def bubble_sort(arr):
    for i in range(len(arr) - 1):
        swapped = False
        for j in range(len(arr) - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break
    return arr

bubble_sort(arr)

File: algorithms/sorting-algorithms/bubble-sort/bubble_sort.jl
arr = [8,2,4,3]

function bubble_sort!(arr::Vector)
    for i in 1:length(arr)-1
        swapped = false
        for j in 1:length(arr)-i
            if arr[j] > arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = true
            end
        end
        !swapped && break
    end
    return arr
end

bubble_sort!(arr)
